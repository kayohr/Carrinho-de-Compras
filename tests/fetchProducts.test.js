require('../mocks/fetchSimulator');
const { fetchProducts } = require('../helpers/fetchProducts');
const computadorSearch = require('../mocks/search');

describe('1 - Teste a função fetchProducts', () => {
  test('Verificando se fetchProducts é uma função', ()=>{
    expect(typeof fetchProducts).toBe('function');
  });
  test('testar se fetch foi chamada com argumento computador', async ()=>{
     fetchProducts('computador');
    expect(fetch).toHaveBeenCalled();
  });
  test('', async ()=>{
    await fetchProducts('computador');
    expect(fetch).toHaveBeenCalledWith('https://api.mercadolibre.com/sites/MLB/search?q=computador');
  });
  test('', async ()=>{
    const result = await fetchProducts('computador');
    expect(result).toEqual(computadorSearch);
  });
  test('', async ()=>{
    const result = await fetchProducts();
    expect(result).toEqual(new Error('You must provide an url'));
  });
  
});





