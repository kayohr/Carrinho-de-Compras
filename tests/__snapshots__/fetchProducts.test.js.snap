// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 - Teste a função fetchProducts : https://api.mercadolibre.com/sites/MLB/search?q=computador 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "https://api.mercadolibre.com/sites/MLB/search?q=computador",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;
